sketch_map_tool/routes.py

```diff
@@ -85,9 +97,22 @@ def status(uuid: str, type_: Literal["quality-report", "sketch-map"]) -> Dict[st
         task = tasks.generate_sketch_map.AsyncResult(task_id)
     else:
         # Unreachable
-        pass
+        raise ValueError
+
+    # see celery states and their precedence here:
+    # https://docs.celeryq.dev/en/stable/_modules/celery/states.html#precedence
+    body = {"id": uuid, "status": task.status, "type": type_}
+    if task.status == SUCCESS:
+        http_status = 200
+        body["href"] = "/api/download/" + uuid + "/" + type_
+    elif task.status in [PENDING, RETRY, RECEIVED, STARTED]:
+        http_status = 202
+    elif task.status in [REJECTED, REVOKED, None, FAILURE]:
+        http_status = 500
+    else:
+        http_status = 500
 
-    return {"id": uuid, "status": task.status}
+    return body, http_status
```

Comment:
```suggestion
    Response(body, status=http_status, mimetype='application/json')
```

Reply:
I find the explicit version better to read. Its somehow assuring to return the Response Object instead just the tuple...